engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -168,6 +172,7 @@ protected Graph readObject(Input in) throws IOException {
     protected void writeObject(Graph obj, Output out) throws IOException {
       out.writeObject(obj.rootScope());
       out.writeInline(scala.collection.immutable.Set.class, obj.links());
+      out.writeInt(obj.nextIdCounter());
```

Comment:
You have to change the `id` of the persisted object to signal that old caches can no longer be read.